ambry-server/src/main/java/com/github/ambry/server/MysqlRepairRequestsDB.java

```diff
@@ -0,0 +1,266 @@
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.sql.DataSource;
+import org.apache.commons.codec.binary.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.utils.Utils.*;
+
+
+/**
+ * RepairRequestsDB.
+ * The class to manipulate the AmbryRepairRequests DB.
+ * The AmbryRepairRequests DB has the records of {@link RepairRequestRecord}.
+ * These are partially failed requests which need to get fixed.
+ */
+public class MysqlRepairRequestsDB {
+  private static final Logger logger = LoggerFactory.getLogger(MysqlRepairRequestsDB.class);
+
+  // table name
+  public static final String REPAIR_REQUESTS_TABLE = "ambry_repair_requests";
+
+  // column names
+  private static final String BLOB_ID = "blobId";
+  private static final String PARTITION_ID = "partitionId";
+  private static final String SOURCE_HOST_NAME = "sourceHostName";
+  private static final String SOURCE_HOST_PORT = "sourceHostPort";
+  private static final String OPERATION_TYPE = "operationType";
+  private static final String OPERATION_TIME = "operationTime";
+  private static final String LIFE_VERSION = "lifeVersion";
+  private static final String EXPIRATION_TYPE = "expirationTime";
+
+  // LOCAL_CONSISTENCY_TODO continue the Get from one token
+  /**
+   * Select the records for one partition with the oldest operation time.
+   */
+  // @formatter:off
+  private static final String GET_QUERY = String.format(""
+      + "SELECT %s, %s, %s, %s, %s, %s, %s, %s "
+      + "FROM %s "
+      + "WHERE %s = ? "
+      + "ORDER BY %s ASC "
+      + "LIMIT 100",
+      BLOB_ID, PARTITION_ID, SOURCE_HOST_NAME, SOURCE_HOST_PORT, OPERATION_TYPE, OPERATION_TIME, LIFE_VERSION, EXPIRATION_TYPE,
+      REPAIR_REQUESTS_TABLE,
+      PARTITION_ID,
+      OPERATION_TIME);
+  // @formatter:on
+
+  /**
+   * Attempt to insert a RepairRequestRecord to the database
+   */
+  // @formatter:off
+  private static final String INSERT_QUERY = String.format(""
+      + "INSERT INTO %1$s "
+      + "(%2$s, %3$s, %4$s, %5$s, %6$s, %7$s, %8$s, %9$s) VALUES (?, ?, ?, ?, ?, ?, ?, ?)",
+      REPAIR_REQUESTS_TABLE,
+      BLOB_ID, PARTITION_ID, SOURCE_HOST_NAME, SOURCE_HOST_PORT, OPERATION_TYPE, OPERATION_TIME, LIFE_VERSION, EXPIRATION_TYPE);
+  // @formatter:on
+
+  /**
+   * Delete a record from the database
+   */
+  // @formatter:off
+  private static final String DELETE_QUERY = String.format(""
+      + "DELETE FROM %s "
+      + "WHERE %s=? AND %s=?",
+      REPAIR_REQUESTS_TABLE,
+      BLOB_ID, OPERATION_TYPE);
+  // @formatter:on
+
+  private final String localDatacenter;
+  private final DataSource dataSource;
+  private final ServerMetrics metrics;
+  private final MySqlUtils.DbEndpoint dbEndpoint;
+  private final Connection connection;
```

Comment:
you shouldn't keep this connection here, since connection might fail, we should get a connection from dataSource every time before running a transaction.

Reply:
You are right. Fixed. Thanks,